Sources/Sentry/SentryNSDataTracker.m

```diff
@@ -0,0 +1,186 @@
+#import "SentryNSDataTracker.h"
+#import "SentryClient+Private.h"
+#import "SentryFileManager.h"
+#import "SentryHub+Private.h"
+#import "SentryLog.h"
+#import "SentryPerformanceTracker.h"
+#import "SentrySDK+Private.h"
+#import "SentryScope+Private.h"
+#import "SentrySpanProtocol.h"
+
+@interface
+SentryNSDataTracker ()
+
+@property (nonatomic, assign) BOOL isEnabled;
+
+@end
+
+@implementation SentryNSDataTracker
+
++ (SentryNSDataTracker *)sharedInstance
+{
+    static SentryNSDataTracker *instance = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{ instance = [[self alloc] init]; });
+    return instance;
+}
```

Comment:
`m`: Does this really have to be a singleton? We only use it in `SentryNSDataSwizzling`, and it could keep a reference to it. Singletons are a pain for testing and I prefer avoiding them.

Reply:
No, it really does not. Just following other integration classes like `SetryNetworkTracker` and `SentryUIViewControllerPerformanceTracker`. Its just a personal preference. Classes that are only a collection of behaviors and hold no state, I prefer to have it singleton instead of creating instances all around the project. But, as you mentioned, this classes are only created once. 

But I would create another PR to remove singletons where they are not needed.

Reply:
Yes, let's please fix it in another PR.